### ISSUE TITLE
UploadDocs failure should throw an Exception

### ISSUE THREAD

**Issue by nginthfs** [fullstorydev/solr-bench#87]
Currently, when UploadDocs fails, we just log the error and go on with the test. This can throw off test results when comparing among Solr setups, since both setups may end up with more/less docs in their collections. To fix this, I think UploadDocs should throw an Exception if it receives a non-200 status code, or at least "fatal" status codes such as 404. Then, callers should handle this by ending the test preemptively (we may not have to change the callers' code, not sure!). 

See: https://github.com/fullstorydev/solr-bench/blob/f8e2567293ff530f2dbf81da9aa57ba4292e2470/src/main/java/org/apache/solr/benchmarks/indexing/UploadDocs.java#L102

**Comment by hiteshk25** [comment]
@nginthfs is there any retry for that?

**Comment by nginthfs** [comment]
I don't believe that there is. I think what happens now is that the doc gets thrown away and the indexing benchmark continues on.

**Comment by patsonluk** [comment]
I know sometimes there could be bad fields in some docs that caused indexing failure, tho that one is not 404, probably 500 i think

**Comment by nginthfs** [comment]
Yeah 500s should probably also fail the task, so we can identify the issue and fix it. The only thing that shouldn't fail the task IMO are timeouts or 429s, which should be retried (although retries are a bigger issue than this, maybe we could add that on as a follow up issue).
